tracer/src/Datadog.Trace/Debugger/Instrumentation/SpanDebuggerInvoker.cs

```diff
@@ -0,0 +1,155 @@
+// <copyright file="SpanDebuggerInvoker.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.TraceAnnotations;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Debugger.Configurations.Models;
+using Datadog.Trace.Debugger.Expressions;
+using Datadog.Trace.Debugger.Instrumentation.Collections;
+using Datadog.Trace.Debugger.RateLimiting;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.Debugger.Instrumentation
+{
+    /// <summary>
+    /// SpanDebuggerInvoker for Async / Non-Async methods.
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class SpanDebuggerInvoker
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(SpanDebuggerInvoker));
+
+        /// <summary>
+        /// [Non Async] Begin Method Invoker
+        /// </summary>
+        /// <param name="probeId">The id of the probe</param>
+        /// <param name="resourceName">The resource name</param>
+        /// <param name="operationName">The operation name</param>
+        /// <returns>Live debugger state</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static SpanDebuggerState BeginSpan(string probeId, string resourceName, string operationName)
```

Comment:
FYI since we have to maintain backwards compatibility, if we need to expand upon the API of `SpanDebuggerInvoker`, we'll have to add new methods and keep these here. For example, we might in the future consider having the user specify a list of tags, and that would need to be a new `BeginSpan` overload

Reply:
Yep, I understand this pain clearly. Once we'll be in GA, we won't touch methods/types that are used by the native instrumentation code.